front/app/modules/commercial/insights/admin/containers/Insights/Details/Network/index.tsx

```diff
@@ -207,7 +214,18 @@ const Network = ({
     } else return true;
   };
 
-  const onZoomEnd = ({ k }: { k: number }) => setZoomLevel(k);
+  const onZoomEnd = ({ k }: { k: number }) => {
+    if (zoomLevel !== k) {
+      if (!initialRender) {
+        setZoomLevel(k);
+        trackEventByName(tracks.zoomVisualization);
+      }
+    } else {
+      if (!initialRender) {
```

Comment:
You could move the `if (!initialRender) `to the outside.

Reply:
Good point!